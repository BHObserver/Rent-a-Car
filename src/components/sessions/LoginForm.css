.login-container {
    max-width: 400px;
    margin: 0 auto;
    padding: 20px;
    border: 1px solid #ccc;
    border-radius: 8px;
  }
  
  h2 {
    font-size: 24px;
    margin-bottom: 20px;
  }
  
  .input-field {
    width: 100%;
    padding: 10px;
    margin-bottom: 15px;
    border: 1px solid #ccc;
    border-radius: 5px;
  }
  
  .login-button {
    width: 100%;
    padding: 10px;
    background-color: #007bff;
    color: #fff;
    border: none;
    border-radius: 5px;
    cursor: pointer;
  }
  
  .login-button:hover {
    background-color: #0056b3;
  }
  
  .error-message {
    color: #ff0000;
    margin-bottom: 10px;
  }
  
  .success-message {
    color: #008000;
    margin-bottom: 10px;
  }
  