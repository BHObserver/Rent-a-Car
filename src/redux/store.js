/* eslint-disable no-unused-vars */
import { configureStore } from '@reduxjs/toolkit';
import authReducer from './reducer/authSlice';
import carReducer from './reducer/carReducer';

const store = configureStore({
  reducer: {
    auth: authReducer,
    car: carReducer, // Add the carReducer under the 'car' key
  },
});

export default store;
