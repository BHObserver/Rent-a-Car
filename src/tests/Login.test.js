import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import LoginForm from '../components/sessions/Login';
import store from '../redux/store';
describe('LoginForm component', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <LoginForm />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
