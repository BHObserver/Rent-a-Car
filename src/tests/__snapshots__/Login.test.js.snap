// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders correctly 1`] = `
<div
  className="background"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <h3>
      Login Here
    </h3>
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      Login
    </button>
    <p>
      Don't have an account yet?
      <a
        href="/signup"
        onClick={[Function]}
      >
        Sign up here.
      </a>
    </p>
  </form>
</div>
`;
